cider/exec/template/CodegenColValues.h

```diff
@@ -72,4 +72,28 @@ class FixedSizeColValues : public NullableColValues {
   DEF_CODEGEN_COL_VALUES_MEMBER(Value, value_)
 };
 
+class MultipleValueColValues : public NullableColValues {
+ public:
+  MultipleValueColValues(std::vector<llvm::Value*> values, llvm::Value* null = nullptr)
+      : NullableColValues(null), values_(values) {}
+  std::unique_ptr<CodegenColValues> copy() const override {
+    return std::make_unique<MultipleValueColValues>(*this);
+  }
+  std::vector<llvm::Value*> getValues() { return values_; }
+  const std::vector<llvm::Value*> getValues() const { return values_; }
+  llvm::Value* getValueAt(int index) { return values_[index]; }
+
+ private:
+  std::vector<llvm::Value*> values_;
+};
+
+class TwoValueColValues : public MultipleValueColValues {
+ public:
+  TwoValueColValues(llvm::Value* value1, llvm::Value* value2, llvm::Value* null = nullptr)
```

Comment:
I think `TwoValueColValues` is not necessary. We can use `MultipleValueColValues` directly with some checks of the `getValues().size() == 3` in varchar case.

Reply:
Of course we can do this check. But if we have some `ThreeValueColValues` or others in the future, we need do such check in different place. it would be simple using `dynamic_cast<TwoValueColValues*>`.

Reply:
Okay. We can keep this for now since not sure whether there are `ThreeValueColValues ` or `FourValueColValues `.